src/main/java/sggw/wzim/czasnawypad/service/FavouriteAttractionService.java

```diff
@@ -32,11 +33,21 @@ public FavouriteAttractionDTO addFavourite(CreateFavouriteAttractionDTO dto) {
 				.orElseThrow(() -> new IllegalArgumentException("Attraction not found"));
 		User user = userRepository.findById(dto.getUserId())
 				.orElseThrow(() -> new IllegalArgumentException(("User not found")));
-		FavouriteAttraction favourite = FavouriteAttraction.builder()
-				.user(user)
-				.attraction(attraction)
-				.build();
-		return toDto(favouriteRepository.save(favourite));	
+		if (favouriteRepository.findByAttractionId(dto.getAttractionId()).isEmpty()) {
+			FavouriteAttraction favourite = FavouriteAttraction.builder()
+					.user(user)
+					.attraction(attraction)
+					.build();
+			return toDto(favouriteRepository.save(favourite));
+		}
+		else
+			return toDto(favouriteRepository.findByAttractionId(dto.getAttractionId()).getFirst());
+
+	}
+
+	public List<FavouriteAttractionDTO> deleteFavourite(Integer toDelete) {
+		favouriteRepository.deleteById(toDelete);
+		return favouriteRepository.findAll().stream().map(this::toDto).collect(Collectors.toList());
 	}

```

Comment:
We don't return anything when deleting the resource. Please drop this implementation and just delete the resource. `void` can be used as a return type.